erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -113,18 +113,22 @@ def get_so_items(self):
 
 		so_list = self.get_so_mr_list("sales_order", "sales_orders")
 
-		item_condition = ""
+		item_condition = variant_of_bom = ""
 		if self.item_code:
+			if frappe.db.exists('Item', self.item_code):
+				has_variant_bom = frappe.db.exists({'doctype': 'BOM', 'item': self.item_code})
+				if not has_variant_bom:
+					variant_of_bom = "'%s'" % frappe.db.get_value('Item', self.item_code, ['variant_of'])
```

Comment:
this exact snippet is used below. If something is repeated make it a reusable function that returns what you need.